hooks/publish.py

```diff
@@ -812,7 +784,22 @@ def _get_publish_symlink_path(self, item, task_settings):
         Extracts the publish symlink path via the configured publish templates
         if possible.
         """
+        publish_symlink_template = task_settings.get("publish_symlink_template").value
+        publish_symlink_path = None
+
+        # If a template is defined, get the publish symlink path from it
+        if publish_symlink_template:
+
+            publish_symlink_path = self._get_path_from_template(publish_symlink_template, item)
+            self.logger.debug(
+                "Used publish_symlink_template to determine the publish path: %s" %
+                (publish_symlink_path,)
+            )
+
+        return publish_symlink_path
+
 
+    def _get_path_from_template(self, template_name, item):
```

Comment:
Separated function to avoid code repetition. Also, subclasses can call this if needed.